src/conversation.rs

```diff
@@ -0,0 +1,155 @@
+use std::ffi::{CStr, CString, NulError};
+use std::ops::{Deref, DerefMut};
+
+use libc::c_void;
+use pam_sys::{set_item, PamHandle, PamItemType};
+
+use crate::{PamError, PamReturnCode};
+
+pub trait ConversationHandler<'a, C>: std::ops::Deref<Target = C> + DerefMut<Target = C> {
+    fn create(conversation: Box<C>, handle: &'a mut PamHandle) -> Self;
+    fn handle_mut(this: &mut Self) -> &mut PamHandle;
+    fn handle(this: &Self) -> &PamHandle;
```

Comment:
Might be my OCD kicking in, but would you swap there two lines? You already implement those two methods in the other order below.